File: VendorResponseSystem/ClientApp/src/models/vendor-response-model.ts
export class VendorResponseModel {
  public constructor(
    id: string,
    user: string,
    buyerCode: string,
    vendCode: string,
    ponumber: string,
    partnumber: string,
    partdescription: string,
    shipqty: string,
    promisedate: string,
    comments: string,
    deliverydate: string,
    quantity: string,
    price: string,
    trackingNumber: string,
    showdeliverydate: boolean = false,
    showquantity: boolean = false,
    showprice: boolean = false,
    showtrackingnumber: boolean = false
  ) {
    this.id = id;
    this.user = user;
    this.buyerCode = buyerCode;
    this.vendCode = vendCode;
    this.ponumber = ponumber;
    this.partnumber = partnumber;
    this.partdescription = partdescription;
    this.shipqty = shipqty;
    this.promisedate = promisedate;
    this.comments = comments;
    this.deliverydate = deliverydate;
    this.quantity = quantity;
    this.price = price;
    this.trackingNumber = trackingNumber;
    this.showdeliverydate = showdeliverydate;
    this.showquantity = showquantity;
    this.showprice = showprice;
    this.showtrackingnumber = showtrackingnumber;
  }
  id!: string;
  user!: string;
  buyerCode!: string;
  vendCode!: string;
  ponumber!: string;
  partnumber!: string;
  partdescription!: string;
  shipqty!: string;
  promisedate!: string;
  comments!: string;
  deliverydate!: string;
  quantity!: string;
  price!: string;
  trackingNumber!: string;
  showdeliverydate!: boolean;
  showquantity!: boolean;
  showprice!: boolean;
  showtrackingnumber!: boolean;
}

File: VendorResponseSystem/ViewModels/VendorResponseModel.cs
namespace VendorResponseSystem.ViewModels
{
    public class VendorResponseModel
    {
        public int Id { get; set; }
        public string User { get; set; }
        public string BuyerCode { get; set; }
        public string VendCode { get; set; }
        public string Ponumber { get; set; }
        public string Partnumber { get; set; }
        public string Partdescription { get; set; }
        public string Shipqty { get; set; }
        public string Promisedate { get; set; }
        public string Comments { get; set; }
        public string Deliverydate { get; set; }
        public decimal Quantity { get; set; }
        public decimal Price { get; set; }
        public string TrackingNumber { get; set; }
        public bool Showdeliverydate { get; set; }
        public bool Showquantity { get; set; }
        public bool Showprice { get; set; }
        public bool Showtrackingnumber { get; set; }
    }
}
